src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,99 @@
+    private User dummyUser;
+
+    @BeforeEach
+    void setUp() {
+        dummyUser = new User();
+        dummyUser.setLogin(DEFAULT_USER_LOGIN_ONE);
+        dummyUser.setPassword(DEFAULT_USER_PASSWORD);
+        dummyUser.setAge(DEFAULT_USER_AGE);
+    }
+
+    @Test
+    void register_alreadyExists_notOk() {
+        storageDao.add(dummyUser);
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(dummyUser));
+        assertEquals("Login already exists", exception.getMessage());
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(null));
+        assertEquals("User cannot be null", exception.getMessage());
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        dummyUser.setLogin(null);
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(dummyUser));
+        assertEquals("Login cannot be null", exception.getMessage());
+    }
+
+    @Test
+    void register_invalidLogin_notOk() {
+        dummyUser.setLogin(SHORT_LOGIN);
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(dummyUser));
+        assertEquals("Login must be at least 6 characters", exception.getMessage());
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        dummyUser.setPassword(null);
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(dummyUser));
+        assertEquals("Password cannot be null", exception.getMessage());
+    }
+
+    @Test
+    void register_invalidPassword_notOk() {
+        dummyUser.setPassword(SHORT_PASSWORD);
+        dummyUser.setLogin(DEFAULT_USER_LOGIN_TWO);
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(dummyUser));
+        assertEquals("Password must be at least 6 characters", exception.getMessage());
+    }
+
+    @Test
+    void register_lowAge_notOk() {
+        dummyUser.setAge(17);
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(dummyUser));
+        assertEquals("Age must be at least 18", exception.getMessage());
+    }
+
+    @Test
+    void register_validUser_Ok() {
+        User expectedUser = new User();
+        expectedUser.setLogin(DEFAULT_USER_LOGIN_ONE);
+        expectedUser.setPassword(DEFAULT_USER_PASSWORD);
+        expectedUser.setAge(DEFAULT_USER_AGE);
+        User actualUser = registrationService.register(expectedUser);
+        assertEquals(expectedUser, actualUser);
+    }
```

Comment:
The test `register_validUser_Ok` doesn't verify if the user was added to storage. Add an assertion to check that the user now exists in the storage after registration.